client/azure.go

```
@@ -141,32 +144,81 @@ func (r *azureRetriever) RetrieveClusterDetailsAndAuthTokens(target Target) (*Ta
 		r.accessToken = oauthToken.AccessToken
 	}
 
-	var client = &http.Client{}
-	client, err := web.NewTLSClient(false, target.CertificateAuthorityData())
-	if err != nil {
-		return nil, fmt.Errorf("unable to create TLS client: %v", err)
-	}
+	var apiServerURL, apiServerCA string
 
-	req, err := createClusterInfoRequest(target.Server())
-	if err != nil {
-		return nil, fmt.Errorf("unable to create cluster-info request: %v", err)
-	}
-	resp, err := client.Do(req)
-	if err != nil {
-		return nil, fmt.Errorf("failed to retrieve cluster-info: %v", err)
-	}
-	clusterInfo, err := pb.ConsumeClusterInfoResponse(resp)
-	if err != nil {
-		return nil, err
+	if target.ShouldFetchCAFromAPIServer() {
+		tlsClient, err := web.NewTLSClient(true)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create TLS client: %v", err)
```

I think this (and many other instances of the same pattern) should use the `%w` error wrapping from go 1.13+: https://go.dev/blog/go1.13-errors.  As part of that, I think stylistically it's considered better not to phrase the error message in terms of an error ("error" or "unable"), since it will eventually be displayed as an error.  It's the difference between:

```
error: unable to frobnicate widget: unable to open thingy: unable to establish wobble: wobble code -1
```

vs

```
error: frobnicating widget: opening thingy: establishing wobble: wobble code -1
```

---

God point. I have updated to use the `%w` format specifier in 6e3a37113153ce76f36d7285431bbd1f6912b759. This also changes the wording from `fmt.Errorf("unable to open..."` to `fmt.Errorf("opening..."`. I liked the idea of this, but the way the default logging presents it, it doesn't read quite as well. What do you think?

```
time="2021-09-23T21:21:43+01:00" level=error msg="Failed to log in to kubectl.local: reading CA response from API Server: my fake error"
time="2021-09-23T21:21:43+01:00" level=fatal msg="Failed to update credentials for some targets."
```

It doesn't scan as well as if it was presented as `error: reading CA response from API Server: my fake error`.

---

I guess rather than logging `Failed to XYZ: %v`, we could log `Error doing XYZ: %v`.  I dunno if it's worth making that further change though -- coherency in error messages isn't _that_ important.

I looked for the original source of the the guidance about error message formatting I passed on above, to see what that had to say.  I read it recently while learning about go errors, but predictably I now can't find it anywhere...

---

I've reduced this part of the change to only replacing `%v` with `%w` in `fmt.Errorf()` calls, so we get the error nesting without rewording things.